console/src/util.rs

```diff
@@ -0,0 +1,28 @@
+pub(crate) trait Percentage {
+    // Using an extension trait for this is maybe a bit excessive, but making it
+    // a method has the nice advantage of making it *really* obvious which is
+    // the total and which is the amount.
+    fn percent_of(self, total: Self) -> f64;
```

Comment:
Does this return the percent out of 100.0, or out of 1.0? Many things assume when it's a float, that it's out of 1.0. It looks like you're making it out of 100 down below. Is there reason to hold onto the float then? Maybe 99.99 percent is nicer to see than 99?

Reply:
we could cast the float back to a usize before returning it. i doubt fractional percentages are terribly valuable here...